### PR TITLE
Feat/add relevant chunk support to colang 2

### PR DESCRIPTION
**Pull request by Pouyanpi**
This PR adds support for the `RetrieveRelevantChunkAction` within Colang 2.0, ensuring that the necessary context is available for the correct functioning of the migrated standard library.

Depends on #705

### CODE REVIEW

File: nemoguardrails/colang/v2_x/library/llm.co [hunk 1]
```diff
@@ -95,6 +95,9 @@ flow continuation on unhandled user utterance
 
   log 'start generating user intent and bot intent/action...'
   $action = 'user said "{$event.final_transcript}"'
+
+  # NOTE: I am not sure if this is needed here.
+  await RetrieveRelevantChunksAction()
```
Comment (drazvan) [inline nemoguardrails/colang/v2_x/library/llm.co:102]: Yes, this is needed here. The relevant chunks need to be computed before the generation happen in `GenerateUserIntentAndBotAction`.

File: nemoguardrails/colang/v2_x/library/llm.co [hunk 2]
```diff
@@ -210,6 +213,8 @@ flow llm generate interaction continuation flow -> $flow_name
   log "generated flow continuation: `{$flow_info}`"
   $exists = await CheckValidFlowExistsAction(flow_id=$flow_info.name)
 
+  await RetrieveRelevantChunksAction()
```
Comment (drazvan) [inline nemoguardrails/colang/v2_x/library/llm.co]: And I think this one needs to be moved above the `await GenerateFlowContinuationAction`

**Review by drazvan — CHANGES_REQUESTED** [review]
@Pouyanpi, we need to add some tests to this as well, to make sure it works and it doesn't break in the future. You can test the right things end up in the prompt with something similar to: https://github.com/NVIDIA/NeMo-Guardrails/blob/develop/tests/test_bug_5.py#L68. 
Thanks!

**Comment by Pouyanpi** [comment]
@drazvan, just made the changes and added a test.

**Review by drazvan — APPROVED** [review]
Thanks! Looks good :+1:.

**Pull request merged**